lib/src/react_client/synthetic_data_transfer.dart

```diff
@@ -104,9 +104,8 @@ SyntheticDataTransfer syntheticDataTransferFactory(Object dt) {
   }
 
   // Copy these lists and ensure they're typed properly.
-  // todo use .cast() in Dart 2
```

Comment:
I gave addressing this a go while I was here 🤷‍♂️  It seemed simple, but I don't fully grasp the why behind the change so let me know if it's actually more complicated or incorrect

Reply:
The reason this code is in place is that we have lists `rawFiles` and `rawTypes` which are `List<dynamic>` (or null), but they need to be converted to `List<File>` and `List<String>`.

The idea of the comment is that in Dart 2 we could start using `cast` like this:
```dart
final files = rawFiles.cast<File>() ?? <File>[];
// or this
final files = (rawFiles ?? []).cast<File>();
// or this lol
final List<File> files = rawFiles?.cast() ?? [];
```

If we had implicit casts disabled in this repo, then the type parameters can be elided if you inline the expression in the constructor argument:

```dart
return new SyntheticDataTransfer(dropEffect, effectAllowed, rawFiles?.cast() ?? [], types?.cast() ?? []);
```
but we don't have em disabled so nevermind haha.

Another option, not using `cast`, would be this:
```dart
final files = <File>[...?rawFiles];
final types = <String>[...?rawTypes];
```
I think that one is my favorite :)

Reply:
Oof, that's clean, thank you for the context and examples! I updated that in 7d1bc70!